### DISCUSSION TITLE
Can anyone tell me about some resources about learning apis i didn't find any good and depth tutorial about creating apis /complex api any resources/books?

### DISCUSSION THREAD

**Discussion by PARKER-X** [fastapi/fastapi#9904] [Show and tell]
Can anyone tell me about some resources about learning apis i didn't find any good and depth tutorial about creating apis /complex api any resources/books?

**Comment by Kludex** [comment]
The documentation is good.

**Reply by PARKER-X** [reply]
No doubt! Fastapi documentation is great!. I am just asking about API design

**Comment by Daschi1** [comment]
Here is a good resource on [Best Practices in API Design](https://swagger.io/resources/articles/best-practices-in-api-design/)

**Reply by PARKER-X** [reply]
Thanks bro!
